_posts/2022-03-05-gs_00006_file_permissions.md

```diff
@@ -0,0 +1,216 @@
+    $ ./test
+
+One tiny note is missing: Who is allowed to change the file permissions? It
+would be stupid if students could change the file permissions of the exam
+themselves, wouldn't it? Only the owner of the file and the **superuser** can
+change it. We talk about the superuser later, but it's basically the owner of
+the machine who can do whatever he wants.
+
+That's it! Easy, isn't it?
+
+# Set permissions using numeric values
+
+Very often there is more than just one way to achieve a goal. There is another
+syntax using numeric values that is commonly used for file permissions. If you
+want to set all permissions (read, write and execute) for the file's owner and
+no permissions for everyone else you can also use the following command:
+
+    $ chmod 700 test
+
+How does this work? Behind the previously used `chmod` command, you see three
+digits. The first represents the owner permissions, the second the permissions
+for the group, and the last for all others. The following table shows the value
+for each permission.
+
+| Permission | Value |
+| ---------- | ----- |
+| Execute    | 1     |
+| Write      | 2     |
+| Read       | 4     |
+
+If you want to combine multiple permissions you just add up the values.
+execute + read + write is represented by 1 + 2 + 4 = 7. The other two values are
+zero since we do not have any permissions for the group and others.
+
+<div class='note-exercise' markdown="1">
+Exercise:
+
+What would be the command for our exam example? Teacher (owner) has read + write
+permissions, participating students (group) has read and everyone else has no
+permissions.
+
+</div>
+
+The solution is:
+
+    $ chmod 640 test
+
+You can decide for yourself which way you prefer. The way from the previous
+section is maybe easier to remember, but it’s completely up to you.
+
+# Summary
+
+Congratulations! Now, you should be able to explain why you get a **permission
+denied** error:
+
+- Use `ls -l`to check the file permissions
+- If you don't have the necessary permission you'll get a _permission denied_
+  error
+- Linux differentiate between **owner**, **group**, and **other**
+- You can **read**, **write** or **execute** a file
+- Use `chmod` to change the permissions
+- One way is to use the abbreviations, e.g. `chmod ugo=wrx test` (u=owner,
+  g=group, o=other, w=write, r=read, x=execute)
+- The other way is to use numerical values like `chmod 640 test`
+
+We leave two small topics as a bonus. We didn't explain how to set the owner or
+group of a file. If you are interested, look for the `chown` command using a
+search engine of your choice or the **man page**. Do you still remember what it
+is? If not, we invite you to read our Terminal Basics posts. Also, we didn't
+explain how to create a group and add users to it. Here the keyword you can
+search for is `groupadd`. However, we probably won't need these commands in our
+upcoming posts.
+
+Yay! Done! This post completes our Linux basics series. The next post starts
```

Comment:
Maybe we could add a quick summary of the series here? But it's not a must.

Reply:
I think about it

Reply:
I think I am going to start with it in the next post